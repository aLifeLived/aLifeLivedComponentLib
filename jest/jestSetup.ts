/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
