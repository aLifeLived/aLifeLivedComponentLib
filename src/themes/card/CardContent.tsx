import React, { ReactNode } from 'react';

// UI
import { Box } from '../box/Box';

type CardContentProps = {
  SVG: ReactNode;
};

export const CardContent: React.FC<CardContentProps> = ({ SVG }) => {
  return (
    <Box alignContent="center" overflow="hidden" borderRadius="large">
      {SVG}
    </Box>
  );
};
