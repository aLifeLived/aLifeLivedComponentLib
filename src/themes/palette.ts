import { opacity } from './utils';

const palette = {
  seeThrew: 'rgba(0,0,0,.5)',

  lightCyan: '#5FD2B5',
  cyan: '#67B6A8', // primary

  yellow: '#FFCB1D',

  blue: '#00B5CD',

  red100: '#D94E69',
  red80: '#F08969', // primary
  red70: '#F7DBC6',
  red05: '#F1ECE5',

  grey100: '#282828',
  grey90: '#606060',
  grey80: '#54575A',
  grey60: '#7F8183',
  grey40: '#A9ABAD',
  grey20: '#CBCBCB',
  grey10: '#EAEAEA',
  grey05: '#FBFBFB',

  black: '#000000',
  white: '#ffffff',
};

export type ColorThemeTokens = keyof typeof ColorThemeLight;

export const ColorThemeLight = {
  primary: palette.cyan,
  primaryOverlay: opacity(palette.cyan, 0.29),
  primaryContrast: palette.grey100,

  secondary: palette.red80,
  secondaryOverlay: opacity(palette.red80, 0.16),

  borderColor: opacity(palette.grey100, 0.12),

  background: palette.grey05,
  backgroundError: palette.red100,
  backgroundContrastHigh: palette.grey100,
  backgroundContrastMid: palette.grey20,
  backgroundContrastLow: palette.white,
  backgroundHighlight: palette.red70,
  backgroundVisible: palette.seeThrew,

  textPrimary: palette.cyan,
  textSecondary: palette.red80,
  textError: palette.red100,
  textContrastHigh: palette.grey100,
  textContrastMid: palette.grey80,
  textContrastLow: palette.grey60,
  textContrastXLow: palette.grey40,
  textContrastXXLow: palette.red05,

  white: palette.white,
  blue: palette.blue,
  yellow: palette.yellow,
  black: palette.black,
  transparent: '#00000000',
};
