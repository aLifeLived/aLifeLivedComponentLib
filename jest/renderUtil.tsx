import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
