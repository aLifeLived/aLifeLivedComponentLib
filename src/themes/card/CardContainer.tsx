import React, { ReactNode } from 'react';

// UI
import { Box } from '../box/Box';
import { ColorThemeTokens } from '../palette';
import { SystemButton } from '../systemButton/SystemButton';

type CardContainerProps = {
  onPress: () => void;
  backgroundColor?: ColorThemeTokens;
  accessibilityLabel?: string;
  testID?: string;
  children: ReactNode;
};

export const CardContainer: React.FC<CardContainerProps> = ({
  onPress,
  backgroundColor = 'backgroundXXLow',
  accessibilityLabel,
  testID,
  children,
}) => {
  return (
    <SystemButton
      onPress={onPress}
      alignContent="space-between"
      borderRadius="large"
      backgroundColor={backgroundColor}
      accessibilityLabel={accessibilityLabel}
      testID={testID}
    >
      <Box>{children}</Box>
    </SystemButton>
  );
};
