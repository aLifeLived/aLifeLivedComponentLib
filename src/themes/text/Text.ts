import { createText, ResponsiveValue } from '@shopify/restyle';
import { ColorThemeTokens } from '../palette';
import { Theme } from '../theme';
import { ValueOf } from '../../ts/utils';

const FontFamilies = {
  sfUiTextBold: 'SFUIText-Bold',
  sfUiTextRegular: 'SFUIText-Regular',
  sfUiTextSemiBold: 'SFUIText-Semibold',
  sfUiDisplayRegular: 'SFUIDisplay-Regular',
};

type TextVariantToken =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'intro'
  | 'body'
  | 'bodySmall'
  | 'bodyXSmall';

type TextVariant = {
  fontFamily: ValueOf<typeof FontFamilies>;
  fontSize: number | ResponsiveValue<number, Theme>;
  lineHeight?: number | string;
  color: ColorThemeTokens;
};

export const textVariants: Record<TextVariantToken, TextVariant> = {
  h1: {
    fontSize: 52,
    lineHeight: 52,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h2: {
    fontSize: 32,
    lineHeight: 40,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h3: {
    fontSize: 24,
    lineHeight: 28.8,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextBold,
  },
  h4: {
    fontSize: 20,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextRegular,
  },
  intro: {
    fontSize: 16,
    lineHeight: 24,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiTextSemiBold,
  },
  body: {
    fontSize: 18,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodySmall: {
    fontSize: 16,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
  bodyXSmall: {
    fontSize: 14,
    lineHeight: 20,
    color: 'textContrastHigh',
    fontFamily: FontFamilies.sfUiDisplayRegular,
  },
};

export const Text = createText<Theme>();
