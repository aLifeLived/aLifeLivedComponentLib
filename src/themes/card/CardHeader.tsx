import React from 'react';

// UI
import { Box } from '../box/Box';
import { Text } from '../text/Text';

type CardHeaderProps = {
  title: string;
};

export const CardHeader: React.FC<CardHeaderProps> = ({ title }) => {
  return (
    <Box paddingRight="m" paddingTop="s" alignSelf="flex-end">
      <Text variant="h4">{title}</Text>
    </Box>
  );
};
