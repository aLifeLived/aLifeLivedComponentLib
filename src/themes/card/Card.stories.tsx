import React from 'react';
import { storiesOf } from '@storybook/react-native';

// SVG
import MyStoriesSVG from '../../assets/svg/my-stories.svg';
import DraftsSVG from '../../assets/svg/drafts.svg';
import TravelThemeSVG from '../../assets/svg/travel-theme.svg';
import COVIDThemeSVG from '../../assets/svg/covid-theme.svg';
import LifeStorySVG from '../../assets/svg/life-story-theme.svg';

// Component
import { CardContainer } from './CardContainer';
import { CardHeader } from './CardHeader';
import { CardContent } from './CardContent';
import { CardOverlay } from './CardOverlay';

// UI
import { Box } from '../box/Box';
import { SystemScrollView } from '../systemScrollView/SystemScrollView';

storiesOf('Cards', module)
  .addDecorator(Story => Story())
  .add('Card with headers', () => (
    <Box flex={1}>
      <Box
        justifyContent="center"
        alignContent="center"
        alignItems="center"
        flexDirection="row"
        flexWrap="wrap"
      >
        <Box marginRight="m" width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="primaryOverlay"
          >
            <CardHeader title="My stories" />
            <CardContent SVG={<MyStoriesSVG height={100} />} />
          </CardContainer>
        </Box>

        <Box marginBottom="m" width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardHeader title="Travel" />
            <CardContent SVG={<TravelThemeSVG height={100} />} />
          </CardContainer>
        </Box>

        <Box marginRight="m" width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="backgroundXXLow"
          >
            <CardHeader title="COVID" />
            <CardContent SVG={<COVIDThemeSVG />} />
          </CardContainer>
        </Box>

        <Box width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="backgroundXXLow"
          >
            <CardHeader title="Drafts" />
            <CardContent SVG={<DraftsSVG height={90} />} />
          </CardContainer>
        </Box>

        <Box width="40%" marginTop="m">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="backgroundXXLow"
          >
            <CardHeader title="Lift Story" />
            <CardContent SVG={<LifeStorySVG height={90} />} />
          </CardContainer>
        </Box>
      </Box>
    </Box>
  ))
  .add('Card with overlay text', () => (
    <SystemScrollView alignContent="center" flex={1}>
      <Box
        flexDirection="row"
        marginLeft="m"
        marginRight="m"
        justifyContent="center"
        flexWrap="wrap"
      >
        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<MyStoriesSVG height={150} width={150} />} />
            <CardOverlay title="School years" />
          </CardContainer>
        </Box>
        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<DraftsSVG height={150} width={150} />} />
            <CardOverlay title="Early life" />
          </CardContainer>
        </Box>

        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<TravelThemeSVG height={150} width={150} />} />
            <CardOverlay title="School years" />
          </CardContainer>
        </Box>

        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<COVIDThemeSVG height={150} width={150} />} />
            <CardOverlay title="Sports" />
          </CardContainer>
        </Box>

        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<LifeStorySVG height={150} width={150} />} />
          </CardContainer>
        </Box>
      </Box>
    </SystemScrollView>
  ));
