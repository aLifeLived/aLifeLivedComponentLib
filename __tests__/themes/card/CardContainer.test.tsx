import React from 'react';
import { render, fireEvent } from '../../../jest/renderUtil';
import { CardContainer } from '../../../src/themes/card/CardContainer';
import { ColorThemeTokens } from '../../../src/themes/palette';
import { Text } from '../../../src/themes/text/Text';

const mockOnPress = jest.fn();
const setup = ({ backgroundColor }: { backgroundColor?: ColorThemeTokens }) =>
  render(
    <CardContainer onPress={mockOnPress} backgroundColor={backgroundColor}>
      <Text>Child</Text>
    </CardContainer>
  );

describe('src >> themes >> card', () => {
  describe('Given background has been passed in as prop', () => {
    test('Should match snapshot', () => {
      const { toJSON } = setup({ backgroundColor: 'blue' });
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Given no background has been passed in as prop', () => {
    test('Should match snapshot', () => {
      const { toJSON } = setup({});
      expect(toJSON()).toMatchSnapshot();
    });
  });

  describe('Given child component has been passed in', () => {
    test('Should render child component', () => {
      const { getByText } = setup({});
      expect(getByText('Child')).toBeDefined();
    });
  });

  describe('Given a user presses a card', () => {
    test('Should fire onPress', () => {
      const { getByText } = setup({});
      fireEvent.press(getByText('Child'));
      expect(mockOnPress).toBeCalledTimes(1);
    });
  });
});
