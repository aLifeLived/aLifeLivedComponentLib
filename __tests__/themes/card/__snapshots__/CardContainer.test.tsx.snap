// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src >> themes >> card Given background has been passed in as prop Should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "space-between",
      "backgroundColor": "#00B5CD",
      "borderRadius": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
          },
        ]
      }
    >
      Child
    </Text>
  </View>
</View>
`;

exports[`src >> themes >> card Given no background has been passed in as prop Should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignContent": "space-between",
      "backgroundColor": "#F1ECE5",
      "borderRadius": 16,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
        },
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "display": "flex",
          },
        ]
      }
    >
      Child
    </Text>
  </View>
</View>
`;
